minio/post_policy.py

```diff
@@ -111,72 +110,52 @@ def set_content_length_range(self, min_length, max_length):
         :param min_length: Minimum length limit for content size.
         :param max_length: Maximum length limit for content size.
         """
-        if min_length > max_length:
-            raise ValueError('minimum limit cannot be larger '
-                             'than maximum limit.')
-        if min_length < 0:
-            raise ValueError('minimum limit cannot be negative.')
-        if max_length < 0:
-            raise ValueError('maximum limit cannot be negative.')
+        err_msg = ('Min-length ({}) must be <= Max-length ({}), '
+                   'and they must be non-negative.').format(
+                       min_length, max_length
+                   )
+        if min_length > max_length or min_length < 0 or max_length < 0:
+            raise ValueError(err_msg)
 
         self._content_length_range = (min_length, max_length)
 
     def _marshal_json(self):
         """
-        Marshal various policies into jsonified byte array.
+        Marshal various policies into json str/bytes.
         """
-        expiration_str = ('"expiration":"' +
-                          self._expiration.strftime("%Y-%m-%dT%H:%M:%S.000Z") +
-                          '"')
-        policies = []
-        for policy in self.policies:
-            policies.append('["' + policy[0] + '","' +
-                            policy[1] + '","' + policy[2] + '"]')
+        policies = self.policies
+        if self._content_length_range:
+            policies.append(['content-length-range'] +
```

Comment:
> The current implementation has all its methods names like set_expires, set_key_startswith, etc but the name set_* implies that if we do the operation repeatedly, only the last value/call has effect - but the implementation does not satisfy this natural expectation - if we call set_bucket_name twice, it will append two conditions to the policy, possibly rendering it invalid - in the S3 spec, it seems these policy conditions are conjunctive (AND-ed together).

To solve this problem we need to fix that first, this has to be done perhaps in all libraries FWICS.